src/main/java/com/cloudbees/jenkins/plugins/mtslavescloud/MansionNodeProvisionerStrategy.java

```diff
@@ -0,0 +1,172 @@
+     * in groovy console
+     *
+     * @param enabled if {@code true} then the strategy will be enabled - including injecting it in the list of
+     *                strategies if necessary, if {@code false} then the strategy will be disabled.
+     */
+    public static void setEnabled(boolean enabled) {
+        MansionNodeProvisionerStrategy strategy = getInstance();
+        if (strategy == null && enabled) {
+            strategy = new MansionNodeProvisionerStrategy();
+            ExtensionList.lookup(NodeProvisioner.Strategy.class).add(0, strategy);
+        }
+        if (strategy != null) {
+            strategy.enabled = enabled;
+        }
+    }
+
+    @Nonnull
+    @Override
+    public NodeProvisioner.StrategyDecision apply(@Nonnull NodeProvisioner.StrategyState strategyState) {
+        final Label label = strategyState.getLabel();
+
+        final SlaveTemplate st = SlaveTemplateList.get().get(label);
+        if (st == null) {
+            return NodeProvisioner.StrategyDecision.CONSULT_REMAINING_STRATEGIES;
+        }
+        if (!st.isEnabled()) {
+            LOGGER.log(Level.FINE, "Slave template is disabled {0}", st);
+            return NodeProvisioner.StrategyDecision.CONSULT_REMAINING_STRATEGIES;
+        }
+
+        final MansionCloud mansionCloud = getCloudImpl();
+        if (mansionCloud == null) {
+            return NodeProvisioner.StrategyDecision.CONSULT_REMAINING_STRATEGIES;
+        }
+        if (mansionCloud.getBackOffCounter(st).isBackOffInEffect()) {
+            LOGGER.log(Level.FINE, "Back off in effect for {0}", st);
+            return NodeProvisioner.StrategyDecision.CONSULT_REMAINING_STRATEGIES;
+        }
+        final HardwareSpec box = mansionCloud.getBoxOf(st, label);
+        if (mansionCloud.getQuotaProblems().isBlocked(box, st)) {
+            LOGGER.log(Level.FINE, "Provisioning of {0} blocked by quota problems.", st);
+            return NodeProvisioner.StrategyDecision.CONSULT_REMAINING_STRATEGIES;
+        }
+        LoadStatistics.LoadStatisticsSnapshot snapshot = strategyState.getSnapshot();
+        int availableCapacity = snapshot.getAvailableExecutors() + snapshot.getConnectingExecutors() +
+                strategyState.getAdditionalPlannedCapacity();
+        int currentDemand = snapshot.getQueueLength();
+        LOGGER.log(Level.FINE, "Available capacity={0}, currentDemand={1}",
+                new Object[]{availableCapacity, currentDemand});
+        if (availableCapacity < currentDemand) {
+            Collection<NodeProvisioner.PlannedNode> plannedNodes = mansionCloud.provision(label, currentDemand - availableCapacity);
+            LOGGER.log(Level.FINE, "Planned {0} new nodes", plannedNodes.size());
+            strategyState.recordPendingLaunches(plannedNodes);
+        }
+        if (availableCapacity < currentDemand) {
+            return NodeProvisioner.StrategyDecision.CONSULT_REMAINING_STRATEGIES;
+        } else {
+            suggestReviewNow(label);
```

Comment:
:bug: don't do this. If you think that further strategies may be interested in provisioning, then return `CONSULT_REMAINING_STRATEGIES`, don't create an infinite loop of recalling this strategy just because it was complete